src/cdomains/unionDomain.ml

```diff
@@ -15,16 +15,29 @@ sig
   val invariant: value_invariant:(offset:Cil.offset -> lval:Cil.lval -> value -> Invariant.t) -> offset:Cil.offset -> lval:Cil.lval -> t -> Invariant.t
 end
 
-module Field =  Lattice.Flat (CilType.Fieldinfo) (struct
-    let top_name = "Unknown field"
-    let bot_name = "If you see this, you are special!"
-  end)
+module Field = struct
+  include Lattice.Flat (CilType.Fieldinfo) (struct
+      let top_name = "Unknown field"
+      let bot_name = "If you see this, you are special!"
+    end)
+
+  let meet f g =
+    if equal f g then
+      f
+    else
+      raise Lattice.Uncomparable
+end
 
 module Simple (Values: Arg) =
 struct
   include Lattice.Prod (Field) (Values)
   type value = Values.t
 
+  let meet (f, x) (g, y) =
+    let field = Field.meet f g in
+    let value = Values.meet x y in
+    (field, value)
+
```

Comment:
Isn't this the default `Lattice.Prod.meet` anyway?

Reply:
This is right, I removed the explicit definition.